compiler/rustc_hir/src/hir.rs

```diff
@@ -524,12 +523,21 @@ pub struct GenericParam<'hir> {
 }
 
 impl GenericParam<'hir> {
-    pub fn bounds_span(&self) -> Option<Span> {
-        self.bounds.iter().fold(None, |span, bound| {
-            let span = span.map(|s| s.to(bound.span())).unwrap_or_else(|| bound.span());
-
-            Some(span)
-        })
+    pub fn bounds_span_for_suggestions(&self) -> Option<Span> {
+        self.bounds
+            .iter()
+            .fold(None, |span: Option<Span>, bound| {
+                // We include bounds that come from a `#[derive(_)]` but point at the user's code,
+                // as we use this method to get a span appropriate for suggestions.
+                // FIXME: a more "principled" approach should be taken here.
+                if !bound.span().can_be_used_for_suggestions() {
```

Comment:
```
let bound_span = bound.span();
bound_span.can_be_used_for_suggestions().then(|| span.map(|s| s.to(bound_span)).unwrap_or(bound_span))
```

Reply:
Also the comment with FIXME is no longer relevant.

Reply:
removed